gradio/components/native_plot.py

```diff
@@ -0,0 +1,254 @@
+from __future__ import annotations
+
+import json
+import warnings
+from typing import TYPE_CHECKING, Any, Callable, Dict, List, Literal
+
+import pandas as pd
+from gradio_client.documentation import document
+
+from gradio.components.base import Component
+from gradio.data_classes import GradioModel
+from gradio.events import Events
+
+if TYPE_CHECKING:
+    from gradio.components import Timer
+
+
+class PlotData(GradioModel):
+    columns: List[str]
+    data: List[List[Any]]
+    datatypes: Dict[str, Literal["quantitative", "nominal", "temporal"]]
+    mark: str
+
+
+class NativePlot(Component):
+    """
+    Creates a native Gradio plot component to display data from a pandas DataFrame. Supports interactivity and updates.
+
+    Demos: native_plots
+    """
+
+    EVENTS = [Events.select]
+
+    def __init__(
+        self,
+        value: pd.DataFrame | Callable | None = None,
+        x: str | None = None,
+        y: str | None = None,
+        *,
+        color: str | None = None,
+        title: str | None = None,
+        x_title: str | None = None,
+        y_title: str | None = None,
+        color_title: str | None = None,
+        x_bin: str | float | None = None,
+        y_aggregate: Literal["sum", "mean", "median", "min", "max", "count"]
+        | None = None,
+        color_map: dict[str, str] | None = None,
+        x_lim: list[float] | None = None,
+        y_lim: list[float] | None = None,
+        caption: str | None = None,
+        sort: Literal["x", "y", "-x", "-y"] | list[str] | None = None,
+        height: int | None = None,
+        label: str | None = None,
+        show_label: bool | None = None,
+        container: bool = True,
+        scale: int | None = None,
+        min_width: int = 160,
+        every: Timer | float | None = None,
+        inputs: Component | list[Component] | set[Component] | None = None,
```

Comment:
See https://github.com/gradio-app/gradio/pull/8854
```suggestion
        inputs: Component | Sequence[Component] | AbstractSet[Component] | None = None,
```